src/compiler/typescript_to_rust/azle_generate/src/candid_type/mod.rs

```diff
@@ -23,12 +23,12 @@ pub mod vec;
 impl TsAst {
     pub fn build_candid_types(&self) -> Result<CandidTypes, Vec<Error>> {
         let (type_aliases, funcs, records, services, tuples, variants) = (
-            self.extract_candid_types(|x| x.to_type_alias()),
-            self.extract_candid_types(|x| x.to_func()),
-            self.extract_candid_types(|x| x.to_record()),
+            self.extract_candid_types(|x| x.to_type_alias(), |p| &p.symbol_table.alias),
```

Comment:
Get rid of this optimization 
Make a ticket with this option in it
add //TODO performance optimization opportunity